ports/psoc6/modules/machine/machine_i2c.c

```diff
@@ -173,13 +203,12 @@ STATIC int machine_i2c_transfer(mp_obj_base_t *self_in, uint16_t addr, size_t le
     }
 }
 
-
 STATIC const mp_machine_i2c_p_t machine_i2c_p = {
+    .deinit = machine_i2c_deinit,
```

Comment:
Not completely sure if this is needed. So far I understand, this is to point to our version of implementation of function, if there is already one implemented in other module (for ex. in extmod/i2c). And I think there is not deinit in extmod. But I might be wrong and feel free to correct.

Reply:
Correct. That was not declared in the extmod. And I added it. 
How can the i2c then be deinited? I wonder how other ports handle softreset: if there is no deinit(), after a softreset the i2c is still configured in whatever mode? If we want to reuse? we reinitiaze? with no impact? 

As discuss, I don´t yet understand how the garbage collector handles destructing all objects... but I don´t see (yet) the connection to hardware peripheral deinitialization. That has to be handled somewhere (freeing before initing() all the time?)...